README.md

```diff
@@ -140,6 +123,35 @@ In order to facilitate easier adoption, script hashes should also be added to CS
+
+
+```
+Content-Security-Policy: script-src 'eval-sha256-SHA256("console.log('Hello World')';
+<script>
+eval("console.log('Hello World')");
+</script>
+```
+
+
+For URL hashes, a similar approach can be followed. Browsers that support the new features can ignore host based allowlisiting if a URL hash is present. Unlike eval though, this leaves some open questions about interactions with other directives, notably `strict-dynamic`. A common fallback for sites that use URL hashes would be to include `https:` in the fallback policy. However sites might require `strict-dynamic` in order to use URL hashes, and they can't set a policy like `script-src https: strict-dynamic, url-sha256-<hash>` since in older browsers `strict-dynamic` would cause `https:` to be [ignored](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy/script-src#strict-dynamic). This can be solved by either:
+
+- Adding a new keyword that enables strict-dynamic like behavior for URL hashes, but is ignored in older browsers, e.g. `strict-dynamic-hashes`.
+- Having the presence of URL hashes imply strict-dynamic, so the keyword is not necessary. This would be the easiest behavior to explain, but it means that sites using hashes lose the option to disable `strict-dynamic` behavior, so sites that don't need it, but want to use URL hashes have a policy that is more lax than they really need.
+- Having the presence of URL hashes imply strict-dynamic, and providing an opt-out keyword, e.g. `no-strict-dynamic-hashes`.
+
+#### Implement the new features as a completely new directive
+The new features can be implemented as part of a separate directive (e.g. `script-src-v2`) that completely replaces `script-src` if it's set. This has the advantage of making the policy easier to read since it's immediately obvious what applies in browsers that are compatible with the new features, and what is the fallback for browsers that are not. On the other hand, the main drawback to this approach is that moves the complexity to the [fallback algorithm](https://www.w3.org/TR/CSP3/#directive-fallback-list).
+
+Both `script-src-elem` and `script-src-attr` fallback directly to `script-src` so they can be handled by either introducing a replacement that supports the new features and replaces the other one if set(e.g. `script-src-elem-v2`), or by completely ignoring them if they are set.
```

Comment:
> or by completely ignoring them if they are set.

or by completely ignoring them if script-src-v2 is set?

Reply:
Done